### PR TITLE
Add "create repo" permission to profile

### PR DESCRIPTION
**Pull request by HelenaEksler**
#12

### CODE REVIEW

File: shoov/shoov.profile [hunk 1]
```diff
@@ -25,6 +25,11 @@ function shoov_install_tasks() {
     'display' => FALSE,
   );
 
+  $tasks['shoov_setup_content_permissions'] = array(
```
Comment (amitaibu) [inline shoov/shoov.profile]: `shoov_setup_content_permissions` => `shoov_setup_permissions`

File: shoov/shoov.profile [hunk 2]
```diff
@@ -77,6 +82,31 @@ function shoov_setup_og_permissions() {
...
 
 /**
+ * Task callback; Setup content permissions.
+ *
+ * We do this here, late enough to make sure all content types were
+ * created.
+ */
+function shoov_setup_content_permissions() {
+  $content_type = 'repository';
+  $content_types = array_keys(node_type_get_types());
```
Comment (amitaibu) [inline shoov/shoov.profile]: I think there no need for all this. I believe we just need the `create repository content` permission. Anyway, Behat will confirm it :)

File: shoov/shoov.profile [hunk 3]
```diff
@@ -77,6 +82,27 @@ function shoov_setup_og_permissions() {
...
+ * We do this here, late enough to make sure all content types were
+ * created.
+ */
+function shoov_setup_permissions() {
+  // Allow user to access messages.
+  // Add repository content permissions.
+  $permissions = array(
+    'create messages',
+    "create repository content",
+    "edit any repository content",
```
Comment (amitaibu) [inline shoov/shoov.profile]: remove

File: shoov/shoov.profile [hunk 4]
```diff
@@ -77,6 +82,27 @@ function shoov_setup_og_permissions() {
...
+ *
+ * We do this here, late enough to make sure all content types were
+ * created.
+ */
+function shoov_setup_permissions() {
+  // Allow user to access messages.
+  // Add repository content permissions.
+  $permissions = array(
+    'create messages',
+    "create repository content",
```
Comment (amitaibu) [inline shoov/shoov.profile]: use `'`

File: shoov/shoov.profile [hunk 5]
```diff
@@ -77,6 +82,27 @@ function shoov_setup_og_permissions() {
 }
 
 /**
+ * Task callback; Setup content permissions.
+ *
+ * We do this here, late enough to make sure all content types were
+ * created.
+ */
+function shoov_setup_permissions() {
+  // Allow user to access messages.
```
Comment (amitaibu) [inline shoov/shoov.profile]: remove.

File: behat/features/repository.feature [hunk 6]
```diff
@@ -15,3 +15,46 @@ Feature: Repository
     | DavidKohav/test-example  |
     | Gizra/Gizra              |
 
+  @api
+  Scenario: Check authenticated user can create a repository
+    Given I login with user "emma"
+    When  I create "Test repository" node of type "repository" with "public" access
```
Comment (amitaibu) [inline behat/features/repository.feature]: no need to check `public`

File: behat/features/repository.feature [hunk 7]
```diff
@@ -15,3 +15,46 @@ Feature: Repository
...
+    Then  I should see the text "Test repository"
```
Comment (amitaibu) [inline behat/features/repository.feature]: `node_access()`

File: behat/features/repository.feature [hunk 8]
```diff
@@ -15,3 +15,46 @@ Feature: Repository
...
+
+  @api
+  Scenario: Check authenticated user can delete a repository
+    Given I login with user "emma"
+    When  I delete "Test repository" node of type "repository"
+    Then  I should see the text "Test repository"
+    And   I should see the text "has been deleted."
+
+  @api
+  Scenario Outline: Check authenticated user without groups can't create nodes of other types.
```
Comment (amitaibu) [inline behat/features/repository.feature:43]: add another scenario, to check can't create content in another group.

**Comment by HelenaEksler** [comment]
@amitaibu Can you review, please?

File: behat/features/bootstrap/FeatureContext.php [hunk 9]
```diff
@@ -67,4 +67,125 @@ public function iVisitNodePageOfType($title, $type) {
...
+   */
+  public function iStartCreatingNodeOfType($type) {
+    $this->getSession()->visit($this->locatePath('node/add/' . $type));
+  }
+
+  /**
+   * @When I create :title node of type :type
+   */
+  public function iCreateNodeOfType($title, $type) {
+    $user = user_load_by_name($this->user->name);
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]: $user => $account

File: behat/features/bootstrap/FeatureContext.php [hunk 10]
```diff
@@ -67,4 +67,125 @@ public function iVisitNodePageOfType($title, $type) {
...
+      ->execute();
+    if (empty($result['node'])) {
+      $params = array(
+        '@title' => $title,
+        '@type' => $type,
+      );
+      throw new \Exception(format_string("Node @title of @type not found.", $params));
+    }
+    $nid = key($result['node']);
+    $this->getSession()->visit($this->locatePath('node/' . $nid . '/delete'));
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]: delete progrmatically.

File: behat/features/bootstrap/FeatureContext.php [hunk 11]
```diff
@@ -67,4 +67,125 @@ public function iVisitNodePageOfType($title, $type) {
...
+    $this->getSession()->getPage()->pressButton('Delete');
+  }
+
+  public function iCheckTheRadioButton($labelText) {
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]: remove

File: behat/features/bootstrap/FeatureContext.php [hunk 12]
```diff
@@ -67,4 +67,125 @@ public function iVisitNodePageOfType($title, $type) {
...
+        return;
+      }
+    }
+    throw new \Exception("Radio button with label {$labelText} not found");
+  }
+
+  /**
+   * @Then I should see :value option in the repositories list
+   */
+  public function iShouldSeeOptionInTheRepositoriesList($value) {
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]: no need, just a node_access()

File: behat/features/bootstrap/FeatureContext.php [hunk 13]
```diff
@@ -67,4 +67,133 @@ public function iVisitNodePageOfType($title, $type) {
...
+    $values = array(
+      'type' => $type,
+      'uid' => $account->uid,
+      'status' => 1,
+      'comment' => 1,
+      'promote' => 0,
+    );
+    $entity = entity_create('node', $values);
+    $wrapper = entity_metadata_wrapper('node', $entity);
+    $wrapper->title->set($title);
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php:138]: `$wrapper->label->set`

File: behat/features/bootstrap/FeatureContext.php [hunk 14]
```diff
@@ -67,4 +67,133 @@ public function iVisitNodePageOfType($title, $type) {
...
+    $entity->field_github_id[LANGUAGE_NONE][0] = array('value' => 123456);
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]: use $wrapper

File: behat/features/bootstrap/FeatureContext.php [hunk 15]
```diff
@@ -67,4 +67,133 @@ public function iVisitNodePageOfType($title, $type) {
...
+   * @Then I should not have access to the page
+   */
+  public function iShouldNotHaveAccessToThePage() {
+    $this->assertSession()->statusCodeEquals('403');
+  }
+
+  /**
+   * @When I start editing :title node of type :type
+   */
+  public function iStartEditingNodeOfType($title, $type) {
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]: remove

File: behat/features/bootstrap/FeatureContext.php [hunk 16]
```diff
@@ -67,4 +67,133 @@ public function iVisitNodePageOfType($title, $type) {
...
+  public function nodeOfTypeShouldBeDeleted($title, $type) {
+    $query = new \entityFieldQuery();
+    $result = $query
+      ->entityCondition('entity_type', 'node')
+      ->entityCondition('bundle', strtolower($type))
+      ->propertyCondition('title', $title)
+      ->propertyCondition('status', NODE_PUBLISHED)
+      ->range(0, 1)
+      ->execute();
+    if (!empty($result['node'])) {
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]:
``` php
if (empty($result['node'])) {
  return;
}
```

File: behat/features/bootstrap/FeatureContext.php (hunk 15 above)
Comment (HelenaEksler) [inline behat/features/bootstrap/FeatureContext.php]: Why remove this function? I'm using this step to check user can edit the repository.

File: behat/features/bootstrap/FeatureContext.php (hunk 13 above)
Comment (HelenaEksler) [inline behat/features/bootstrap/FeatureContext.php:138]: Can't use `$wrapper->label->set()`, getting an error. As I understand, for entity metadata wrapper there is only `label()` that returns label or title of the entity. But to set the value we should use the `title` name of the field.

**Comment by HelenaEksler** [comment]
@amitaibu Ready.

File: behat/features/bootstrap/FeatureContext.php [hunk 17]
```diff
@@ -67,4 +67,138 @@ public function iVisitNodePageOfType($title, $type) {
...
+      $params = array(
+        '@title' => $title,
+        '@type' => 'repository',
+      );
+      throw new \Exception(format_string("Node @title of @type not found.", $params));
+    }
+    $nid = key($result['node']);
+    $account = user_load_by_name($this->user->name);
+    if (og_user_access('node', $nid, 'access group', $account)) {
+      throw new \Exception(format_string("You can add content to @title group", array('@title' => $title)));
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]: `You` => `User @name`

File: behat/features/bootstrap/FeatureContext.php [hunk 18]
```diff
@@ -67,4 +67,138 @@ public function iVisitNodePageOfType($title, $type) {
...
+    if (empty($result['node'])) {
+      $params = array(
+        '@title' => $title,
+        '@type' => 'repository',
+      );
+      throw new \Exception(format_string("Node @title of @type not found.", $params));
+    }
+    $nid = key($result['node']);
+    $account = user_load_by_name($this->user->name);
+    if (og_user_access('node', $nid, 'access group', $account)) {
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]: this is actually wrong -- there's no such permission defined by og. it should be enough to node_access() the group.

File: behat/features/bootstrap/FeatureContext.php [hunk 19]
```diff
@@ -67,4 +67,138 @@ public function iVisitNodePageOfType($title, $type) {
...
+
+  /**
+   * @When I create :title node of type :type
+   */
+  public function iCreateNodeOfType($title, $type) {
+    $account = user_load_by_name($this->user->name);
+    $values = array(
+      'type' => $type,
+      'uid' => $account->uid,
+      'status' => 1,
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]: can we remove this and other properties -- I think you get them by default.

File: behat/features/bootstrap/FeatureContext.php [hunk 20]
```diff
@@ -67,4 +67,138 @@ public function iVisitNodePageOfType($title, $type) {
...
+      '@type' => $type,
+    );
+    throw new \Exception(format_string("You can't edit the node @title of @type not found.", $params));
+  }
+
+  /**
+   * @When I start creating node of type :type
+   */
+  public function iStartCreatingNodeOfType($type) {
+    $this->getSession()->visit($this->locatePath('node/add/' . $type));
```
Comment (amitaibu) [inline behat/features/bootstrap/FeatureContext.php]: node_access() instead

**Comment by amitaibu** [comment]
Merged, thanks.

**Pull request closed without merging**